### PR TITLE
Forces the update of the rendered_text attribute when closing the overlay

### LINKED ISSUE #2120: Image render funkyness from Table -> Visualization views
I made a quick [Rosa Parks map](https://team.cartodb.com/u/aureliamoser/viz/5f164106-a8c3-11e4-ae4a-0e018d66dc29/map) to demo to journalists; the visualization view doesn't pull the images properly, but I can see them in the edit interface. 

For context, 
* I have Odyssey enabled in my account.
* I loaded in a table and build my slides in the map view of the editor. 
* When I publish to visualization, the images render as broken grey boxes.

I'm using two tables [rosaparks here](https://team.cartodb.com/u/aureliamoser/tables/rosaparks/map) and [freedomroute points here](https://team.cartodb.com/u/aureliamoser/tables/freedomroute) and then another [freedomroute_line](https://team.cartodb.com/u/aureliamoser/tables/freedomroute_line) where ST-makeline between the points so there's a "route"; not sure if the number of layers affects anything, but maybe it does.

This is what I see in the editor:
screen shot 2015-02-06 at 6 16 34 pm
This is what the Visualization displays:
screen shot 2015-02-06 at 6 16 57 pm

### PR DESCRIPTION
**Pull request by javierarce**
Fixes #2120 #2020 #1979.

@CartoDB/frontend: review this, please.

### CODE REVIEW

**Comment by Cartofante** [comment]
Frontend tests were OK :+1: ([details](http://clinker.cartodb.net/jenkins/job/CartoDB-PR-testing/1455/))

File: lib/assets/javascripts/cartodb/table/overlays/image.js [hunk 1]
```diff
@@ -206,10 +206,10 @@ cdb.admin.overlays.Image = cdb.admin.overlays.Text.extend({
     this.$text.html(img);
     this.$text.attr("contenteditable", false);
 
-    var self = this;
-
     extra.url = url;
 
+    extra.rendered_text = "<img src='" + url + "' style='width: " + boxWidth + "px'/>";
+
     if (extra.url !== extra.default_image_url) extra.has_default_image = false;
```
Comment (xavijam) [inline lib/assets/javascripts/cartodb/table/overlays/image.js]: I'm trying to follow JS guide lines we described [here](https://github.com/Vizzuality/cartodb-management/wiki/CartoDB-JavaScript-Style-Guide), I encourage @CartoDB/frontend to try from now on.
For example, for inline ifs -> https://github.com/Vizzuality/cartodb-management/wiki/CartoDB-JavaScript-Style-Guide#conditional-expressions--equality

File: lib/assets/javascripts/cartodb/table/overlays/image.js [hunk 2]
```diff
@@ -266,6 +266,7 @@ cdb.admin.overlays.Image = cdb.admin.overlays.Text.extend({
     var img       = "<img src='" + url + "' style='width: " + boxWidth + "px'/>";
```
Comment (xavijam) [inline lib/assets/javascripts/cartodb/table/overlays/image.js]: I'd create a  template if it is used several times.

**Comment by xavijam** [comment]
I've checked there is a bind over document, is it cleaned when the view executes the clean method?

File: lib/assets/javascripts/cartodb/table/overlays/image.js [hunk 3]
```diff
@@ -329,10 +330,8 @@ cdb.admin.overlays.Image = cdb.admin.overlays.Text.extend({
 
     this.$text = this.$el.find(".content div.text");
```
Comment (xavijam) [inline lib/assets/javascripts/cartodb/table/overlays/image.js]: this.$(".content div.text");

**Comment by xavijam** [comment]
No more comments about the changes :+1:.

**Comment by Cartofante** [comment]
Frontend tests were OK :+1: ([details](http://clinker.cartodb.net/jenkins/job/CartoDB-PR-testing/1459/))

**Comment by Cartofante** [comment]
Frontend tests were OK :+1: ([details](http://clinker.cartodb.net/jenkins/job/CartoDB-PR-testing/1461/))

**Comment by Cartofante** [comment]
Frontend tests were OK :+1: ([details](http://clinker.cartodb.net/jenkins/job/CartoDB-PR-testing/1460/))

**Comment by Cartofante** [comment]
Frontend tests were OK :+1: ([details](http://clinker.cartodb.net/jenkins/job/CartoDB-PR-testing/1462/))

**Pull request merged**